File: VB/ConsoleApp/Shapes/Textbox.vb
Imports ConsoleApp.Printers

Namespace Shapes

    Public Class Textbox
        Inherits Shape
        Implements IPrintable

        Public Property Text As String = String.Empty
        Public Property Width As Integer = 0
        Public Property Height As Integer = 0

        Public Sub New(x As Integer, y As Integer, width As Integer, height As Integer, Optional text As String = "")
            Me.X = x
            Me.Y = y
            Me.Width = width
            Me.Height = height
            Me.Text = text
        End Sub

        Public Overrides Function ToString() As String Implements IPrintable.ToString
            Return String.Format("Textbox ({0},{1}) width={2} height={3} Text=""{4}""", X, Y, Width, Height, Text)
        End Function

    End Class

End Namespace

File: CSharp/ConsoleApp/ConsoleApp/Concrete/Shapes/Textbox.cs
using ConsoleApp.Abstractions;

namespace ConsoleApp.Concrete.Shapes
{
    public class Textbox : Shape, IPrint
    {
        public int Width { get; set; }
        public int Height { get; set; }
        public string Text { get; set; }

        public Textbox(int x = 5, int y = 5, int width = 200, int height= 100, string text = @"sample text")
        {
            this.X = x;
            this.Y = y;
            this.Width = width;
            this.Height = height;
            this.Text = text;
        }

        public override string ToString()
        {
            return string.Format(@"Textbox ({0},{1}) width={2} height={3} Text=""{4}""", X, Y, Width, Height, Text);
        }
    }
}
